api/v1beta1/common_types.go

```diff
@@ -49,6 +49,19 @@ type BarbicanTemplate struct {
 	// +kubebuilder:validation:Required
 	// ServiceAccount - service account name used internally to provide Barbican services the default SA name
 	ServiceAccount string `json:"serviceAccount"`
+
+        // +kubebuilder:validation:Optional
+        PKCS11 *BarbicanPKCS11Template `json:"pkcs11,omitempty"`
```

Comment:
So this is a new optional struct field. 
If the user is not passing anything here then the field will be defaulted to nil and therefore the required struct fields does not cause a problem as the struct is not instantiated. 
If the user is passing a value to this field then the user will need to provide at least all the required struct fields as well. 

This feels like an exception over the rule of not adding new required fields, as the struct field wrapping the new required field is i) introduced by the same change so there is no CRs on the field with a value for this field ii) the struct field is introduced as an optional, *struct field, defaulted to nil.

I suggest to keep the crd check in place but ignore this false positive when merging the PR (by overriding the CI result) if the rest of the pre-commit checks are passing.